pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -560,6 +562,32 @@ func (p *DNSProxy) CheckAllowed(endpointID uint64, destPort uint16, destID ident
 	return false, nil
 }
 
+func configureConnection(conn *net.Conn, sec_id identity.NumericIdentity) error {
+
+	var file *os.File
+	var err error
+
+	switch l4conn := (*conn).(type) {
+	case *net.UDPConn:
+		if file, err = l4conn.File(); err != nil {
+			return fmt.Errorf("can't get file from %v: %v", l4conn, err)
+		}
+	case *net.TCPConn:
+		if file, err = l4conn.File(); err != nil {
+			return fmt.Errorf("can't get file from %v: %v", l4conn, err)
+		}
+	default:
+		return fmt.Errorf("unknown type %T", l4conn)
+	}
+
+	mark := int(uint32(sec_id)<<16 | uint32(linux_defaults.MagicMarkIsProxy))
+	err = unix.SetsockoptInt(int(file.Fd()), unix.SOL_SOCKET, unix.SO_MARK, mark)
```

Comment:
Will this need to depend on `option.Config.EnableIdentityMark`?

Reply:
As far as I see, no. This option is only used in `from-container` attached to `lxc*`, and the traffic in patch originates in the host namespace